### ISSUE TITLE
Python scripts in prompts does not work with anthropic as provider

### ISSUE THREAD

**Issue by harsh-writesonic** [promptfoo/promptfoo#782]
Stacktrace:
```
/usr/local/lib/node_modules/promptfoo/dist/src/prompts.js:225
                            args: [JSON.stringify(context)],
                                        ^

TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Anthropic'
    |     property 'completions' -> object with constructor 'Completions'
    --- property '_client' closes the circle
    at JSON.stringify (<anonymous>)
   ```
  
promptfooconfig.yaml
```
prompts:
  - file://prompt.py

providers:
  - openai:gpt-4o
  - anthropic:messages:claude-3-opus-20240229
  ``` 
   
Nodejs version - v20.12.2
Promptfoo version - 0.58.1
Please let me know if it is reproducible for you.

**Comment by typpo** [comment]
Thanks for flagging!  Fix in https://github.com/promptfoo/promptfoo/pull/784

As a temporary workaround, you can switch to using a specific function in the python file, e.g. `prompt.py:my_prompt`.  That should work alright in the current version.

**Comment by harsh-writesonic** [comment]
Thanks for quick fix.

**Comment by typpo** [comment]
Released in 0.59.0!
